### DISCUSSION TITLE
how do i grep all open buffers?

### DISCUSSION THREAD

**Discussion by AlejandroSanchez90** [ibhagwan/fzf-lua#1531] [Q&A] [answered]
is there a way to grep on all open buffers? something like FzfLua grep_curbuf but using all buffers

**Comment by ibhagwan** [comment]
Send all open buffers to quick fix and use `grep_quickfix`.

**Reply by AlejandroSanchez90** [reply]
while we are on a quick fix, is there a way to open troubles quick fix list instead of the Nvim one ?

**Reply by ibhagwan** [reply]
> while we are on a quick fix, is there a way to open troubles quick fix list instead of the Nvim one ?

I’m not familiar with the internals of trouble, but if it’s not a neovim quick fix you’d have to use the API for a custom picker or ask upstream if there’s already an integration.

**Reply by AlejandroSanchez90** [reply]
trying your suggestion for grepping using all buffers, when i open buffers pick to send to quickfix, the active one is disable and its not been picked up by crtl + q ("select-all+accept")

**Reply by ibhagwan** [reply]
> trying your suggestion for grepping using all buffers, when i open buffers pick to send to quickfix, the active one is disable and its not been picked up by crtl + q ("select-all+accept")

Disable sorting and it will be enabled, `:FzfLua buffers sort_lastused=false`.

**Reply by ibhagwan** [reply]
If you do this a lot, I’d skip the call to `:FzfLua buffers`,  create a dedicated function/keybind that calls `nvim_list_bufs()`, sends them all to the quick fix (few lines of code) and calls `grep_quickfix`.

**Reply by AlejandroSanchez90** [reply]
will do that thanks!

**Comment by AlejandroSanchez90** [comment, marked as answer]
for anyone interested this is the function im using for this
```lua
function SendBuffersToQuickfix() 
 local bufs = vim.api.nvim_list_bufs() 
 local qflist = {} 
 for _, buf in ipairs(bufs) do 
  table.insert(qflist, { bufnr = buf }) 
 end 
 vim.fn.setqflist(qflist, "r") 
 fzflua.grep_quickfix() 
end
```
